src/utilities/linkHighlightedJson.js

```diff
@@ -0,0 +1,83 @@
+import _ from 'lodash';
+import {xdrViewer, singleAccount} from './linkBuilder';
+
+// Map of functions specific to JSON property keys that creates urls for a given
+// value.
+// A function can also return undefined if it doesn't deem the value worthy of
+// linking (for example: invalid account ID or keys that have ambiguous names)
+let accountIdGenerator = (value) => {
+  if (value.length === 56) {
+    return singleAccount(value);
+  }
+};
+let valueUrlGenerator = {
+  href: (value) => {
+    return value;
+  },
+  'envelope_xdr': (value) => {
+    return xdrViewer(value, 'TransactionEnvelope');
+  },
+  'result_xdr': (value) => {
+    return xdrViewer(value, 'TransactionResult');
+  },
+  'result_meta_xdr': (value) => {
+    return xdrViewer(value, 'TransactionMeta');
+  },
+  'id': accountIdGenerator,
+  'public_key': accountIdGenerator,
+  'account_id': accountIdGenerator,
+  'funder': accountIdGenerator,
+  'account': accountIdGenerator,
+  'source_account': accountIdGenerator,
+  'destination_account': accountIdGenerator,
+};
+
+// linkHighlightedJson annotates and adds relevant links to json code highlighted
+// by prism.js.
+
+// @param {string} code - Valid html generated by prism.js in JSON language mode
+export default function linkHighlightedJson(code) {
+  let parser = new DOMParser();
+  let document = parser.parseFromString(code, 'text/html');
+  let body = document.childNodes[0].childNodes[1];
+
+  // Edit the token elements in place
+  _.each(body.children, (token, index) => {
+    if (token.className.indexOf('property') > 0) {
+      annotatePropertyToken(token);
+    }
+  });
+  return body.innerHTML;
+}
+
+// Takes in a property highlighted item. If functionality is defined for it's
+// corresponding property (only string type supported) and the content is valid,
+// then we will turn the value node into a link.
+// @param {HTMLElement} propertyToken - The token element of property type (assigned by prism)
+function annotatePropertyToken(propertyToken) {
+  let valueToken = propertyToken.nextElementSibling.nextElementSibling;
```

Comment:
Parsing the HTML using the DOMParser isn't brittle;  I was specifically referring to how you are looking up `valueToken`.  

Relying on the relative structure of the dom to extract elements is brittle, which is to say that any change in the relationship between `propertyToken`  and `valueToken` will break the above code.  Prism gets updates and adds a new nesting? This breaks.  Prism gets updated and inserts another intervening sibling element? This breaks.

Instead, you should strive to use targets that are not dependent upon the full relationship between two nodes.  Are there any identifying classes on `valueToken` that can be used to find it?  Is there no other way to identify `valueToken` beyond "The second sibling element of `propertyToken"?

At the very least, you should raise a custom error notifying the developer that the expected dom structure is no longer reality so that they don't get a "Cannot read property 'nextSiblingElement' of undefined" error.